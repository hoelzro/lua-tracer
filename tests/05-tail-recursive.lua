local function length(list, accum)
  accum = accum or 0

  if not list then -- trace: accum
    return accum
  else
    return length(list.next, accum + 1)
  end
end

local list = {}

do
  local node = list

  for i = 1, 5 do
    node.next = {}
    node = node.next
  end
end

length(list)
