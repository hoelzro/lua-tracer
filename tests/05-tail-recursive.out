accum	0
accum	1
accum	2
accum	3
accum	4
accum	5
accum	6
